camel/CSVExample/src/main/java/ibmi/example/CSVExample.java

```diff
@@ -0,0 +1,178 @@
+ */
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.springframework.stereotype.Component;
+import org.apache.camel.dataformat.csv.CsvDataFormat;
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** 
+ * IBM i specific imported libraries for connecting to the DB2 database
+ */
+import com.ibm.as400.access.AS400JDBCDataSource;
+
+/**
+ * This is a CSV example for creating a CSV from a query and doing different operations with it as a result
+ */
+public class CSVExample {
+
+    public static void main(final String... args) throws Exception {
+
+        /**
+         * Values for configuring application. Setting constants at the top of the app for configuration 
+         * and then using variables within the app allow you to configure your app for multiple situations
+         * without having to dig through code and make changes within the code. 
+         * 
+         * Options are:
+         * standardcsvexample - Just creating a standard CSV, no options
+         * pipedelimitedexample - Creating a pipe deliminited example and removing double quotes from strings
+         * displayindividualfieldoutput - Selecting just the first row from the query results, and outputting
+         * to the screen, no file written
+         * 
+         */
+        final String whatexample = "standardcsvexample";
+
+        /**
+         * This is the location we want our file stored in after it is created. Note that you can start from 
+         * root with a leading forward slash, or a subdirectory of where the program is called like the example
+         * below
+         */
+        final String filelocation = "tmp/";
+
+
+        /**
+         * Standard for a Camel deployment. Start by getting a CamelContext object.
+         */
+        CamelContext context = new DefaultCamelContext();
+
+        /**
+         * This sets up the connection for local IBM i Here is where we will store
+         * connection information for the application
+         */
+        AS400JDBCDataSource localDS = new AS400JDBCDataSource("localhost", "*CURRENT", "*CURRENT");
+
+        /**
+         * These values below exist if you are not journaling your IBM i table that you are querying
+         */
+        //localDS.setTransactionIsolation("none");
+        
+        /**
+         * This binds the localDS we set above to the route jt400
+         */
+        context.getRegistry().bind("jt400", localDS);
+
+        /**
+         * Setting our Marshall options for CSV
+         */
+		
+
+        /**
+         * What example do we want to accomplish? Set the variable above to choose the example 
+         */
+        context.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() {
+                from("timer://ibmiexamplecheckinterval?period=5000")
```

Comment:
I'd recommend `synchronous=true` on this one, in case there are delays in the QSYS2 service

Reply:
What does that mean and how does it work?

Reply:
Turns on synchronous mode, which _I believe_ ensures the route won't kick off until the previous invocation of the route ends. That is, if you have a timer firing every 5 seconds and are running a 20-second task in asynchronous mode, you could end up with the same route running 4/5 times at any given moment (add more for possible fluctuations, contention, etc).